code/go/0chain.net/blobbercore/challenge/challenge.go

```diff
@@ -70,12 +71,14 @@ func saveNewChallenge(c *ChallengeEntity, ctx context.Context) {
 		}
 	}()
 
-	db := datastore.GetStore().GetDB()
-	if Exists(db, c.ChallengeID) {
+	ctx = datastore.GetStore().CreateTransaction(ctx)
+	tx := datastore.GetStore().GetTransaction(ctx)
+
+	if Exists(tx, c.ChallengeID) {
 		return
```

Comment:
it makes transaction is never closed if challenge exists on db. [db.Transaction](https://gorm.io/docs/transactions.html#Transaction) is better.

Reply:
Yes this was the reason why tests were failing.
Fixed it with defer function